nilearn/datasets/func.py

```diff
@@ -1276,3 +1276,189 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
+    interpreted differently, so it is important that you consult with your IRB
+    or Ethics Committee before beginning your research. If needed and upon request,
+    the HCP will provide a certificate stating that you have accepted the
+    HCP Open Access Data Use Terms. Please note that everyone who works with
+    HCP open access data must review and agree to these terms, including those
+    who are accessing shared copies of this data. If you are sharing
+    HCP Open Access data, please advise your co-researchers that they must
+    register with ConnectomeDB and agree to these terms.
+    Register and sign the Open Access Data Use Terms at
+    ConnectomeDB: https://db.humanconnectome.org/
+    """
+    url = "https://www.nitrc.org/frs/download.php/8037/Megatrawls.tgz"
+    opts = {'uncompress': True}
+
+    # dataset terms
+    dimensionalities = [25, 50, 100, 200, 300]
+    timeseries_methods = ['multiple_spatial_regression', 'eigen_regression']
+
+    message = ("The %s you have given '%s' is invalid. Please choose either "
+               "of them %s or list of specific choices.")
+
+    error_correcting_names = ['dimensionality', 'timeseries']
+    user_inputs = [dimensionality, timeseries]
+    assign_names = ['dimensionalities', 'timeseries_methods']
+    standard_variables = [dimensionalities, timeseries_methods]
+
+    for name, check_in, assign, standard in zip(error_correcting_names, user_inputs,
+                                                assign_names, standard_variables):
+        if check_in is not None:
+            if isinstance(check_in, list):
+                for each_str in check_in:
+                    if each_str not in standard:
+                        raise ValueError(message % (
+                            name, check_in, str(standard)))
+                if assign == 'dimensionalities':
+                    dimensionalities = check_in
+                else:
+                    timeseries_methods = check_in
+            elif not isinstance(check_in, list):
+                if check_in not in standard:
+                    raise ValueError(message % (
+                        name, check_in, str(standard)))
+
+                if assign == 'dimensionalities':
+                    dimensionalities = [check_in]
+                else:
+                    timeseries_methods = [check_in]
+
+    n_combinations = len(dimensionalities) * len(timeseries_methods)
+    dataset_name = 'Megatrawls'
+    files_netmats1 = []
+    files_netmats2 = []
+    for dim in dimensionalities:
+        filename = os.path.join('3T_Q1-Q6related468_MSMsulc_d' + str(dim))
+        for timeseries in timeseries_methods:
+            if timeseries == 'multiple_spatial_regression':
+                timeseries = 'ts2'
+            elif timeseries == 'eigen_regression':
+                timeseries = 'ts3'
+            each_files_netmats1 = [(os.path.join(
+                filename + '_' + str(timeseries), 'Znet1.txt'), url, opts)]
```

Comment:
Why don't you give the user the choice between `correlation` and `partial_correlation`?